File: server/apps/nexotype/schemas/clinical/indication_schemas.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime

# ==========================================
# Request Schemas
# ==========================================

class IndicationCreate(BaseModel):
    """Schema for creating a new indication"""
    name: str
    icd_10_code: str | None = None
    meddra_id: str | None = None


class IndicationUpdate(BaseModel):
    """Schema for updating an indication"""
    name: str = None
    icd_10_code: str | None = None
    meddra_id: str | None = None


# ==========================================
# Response Schemas
# ==========================================

class IndicationDetail(BaseModel):
    """Schema for indication details"""
    id: int
    name: str
    icd_10_code: str | None = None
    meddra_id: str | None = None
    created_at: datetime
    updated_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class IndicationListResponse(BaseModel):
    """Response schema for listing indications"""
    success: bool
    data: list[IndicationDetail] | None = None
    count: int = 0
    error: str | None = None


class IndicationResponse(BaseModel):
    """Response schema for single indication operations"""
    success: bool
    data: IndicationDetail | None = None
    error: str | None = None


class MessageResponse(BaseModel):
    """Simple message response"""
    success: bool
    message: str | None = None
    error: str | None = None

File: client/src/modules/nexotype/schemas/clinical/indication.schemas.ts
import { z } from 'zod';

// ==========================================
// Full Schema (GET responses)
// ==========================================

export const IndicationSchema = z.object({
  id: z.number(),
  name: z.string().min(1).max(255),
  icd_10_code: z.string().max(20).nullable().optional(),
  meddra_id: z.string().max(20).nullable().optional(),
  created_at: z.string(),
  updated_at: z.string().nullable(),
});

// ==========================================
// Input Schemas (POST/PUT)
// ==========================================

export const CreateIndicationSchema = z.object({
  name: z.string().min(1).max(255),
  icd_10_code: z.string().max(20).nullable().optional(),
  meddra_id: z.string().max(20).nullable().optional(),
});

export const UpdateIndicationSchema = z.object({
  name: z.string().min(1).max(255).optional(),
  icd_10_code: z.string().max(20).nullable().optional(),
  meddra_id: z.string().max(20).nullable().optional(),
});

// ==========================================
// Types
// ==========================================

export type Indication = z.infer<typeof IndicationSchema>;
export type CreateIndication = z.infer<typeof CreateIndicationSchema>;
export type UpdateIndication = z.infer<typeof UpdateIndicationSchema>;

// ==========================================
// Response Types
// ==========================================

export type IndicationResponse = {
  success: boolean;
  data?: Indication;
  error?: string;
};

export type IndicationsResponse = {
  success: boolean;
  data?: Indication[];
  count?: number;
  error?: string;
};
